### ISSUE TITLE
Using ListParams in BankAccountListParams results in unexpected query params in URL

### ISSUE THREAD

**Issue by sfriedel** [stripe/stripe-go#632]
While mocking part of the Stripe API for use in our test suite I was noticing that setting `ListParams` in `BankAccountListParams` produces unexpected and possibly invalid query params in the requested URL.

I prepared a short test program to demonstrate the issue:

```go
package main

import (
	"fmt"
	"net/http"
	"net/http/httptest"

	"github.com/stripe/stripe-go"
	"github.com/stripe/stripe-go/client"
)

func main() {
	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		fmt.Println("URI:", r.RequestURI)
		fmt.Println("QUERY:", r.URL.Query())
		w.WriteHeader(500)
	}))
	backends := &stripe.Backends{
		API: &stripe.BackendConfiguration{
			Type:       stripe.APIBackend,
			URL:        server.URL,
			HTTPClient: http.DefaultClient,
		},
		Uploads: &stripe.BackendConfiguration{
			Type:       stripe.UploadsBackend,
			URL:        server.URL,
			HTTPClient: http.DefaultClient,
		},
	}
	c := client.New("sk_xxxxx", backends)
	p := &stripe.BankAccountListParams{
		Account: stripe.String("acc_xxxxxx"),
		ListParams: stripe.ListParams{
			StartingAfter: stripe.String("ba_xxxxxx"),
			Limit:         stripe.Int64(1),
		},
	}
	c.BankAccounts.List(p)
}
```

When this is run, it will produce the following output:

```
$ go run test.go
URI: /accounts/acc_xxxxxx/external_accounts?object=bank_account?limit=1&starting_after=ba_xxxxxx
QUERY: map[object:[bank_account?limit=1] starting_after:[ba_xxxxxx]]
```

As can be seen in the QUERY output, the `object` param actually also contains the `limit` because they are separated by `?` instead of `&`.

It does not seem to cause any issues when used with the regular Stripe API but it might be undesired behavior anyway.

**Comment by brandur-stripe** [comment]
Thank you for the report @sfriedel! Your repro script was very handed too — I ended up using it to verify that my fix actually worked.

Addressed in #633.

**Comment by brandur-stripe** [comment]
Fix released as 36.1.2.

**Comment by sfriedel** [comment]
Thank you @brandur-stripe
